### ISSUE TITLE
Getting 404s when trying to assign an SLA domain to an existing VM

### ISSUE THREAD

**Issue by bender-the-greatest** [rubrikinc/rubrik-plugin-for-vrealize#11]
**Expected Behavior**

Please describe the behavior you are expecting.
When calling the "Rubrik - Assign SLA Domain" workflow, it assigns the SLA Domain to the target VM.

**Current Behavior**

What is the current behavior?

When trying to assign an SLA domain during the PRE Machine Provisioned state within vRO, the "Rubrik - Assign SLA Domain" workflow hangs on the rubrik_SetSLAByVmId action. If I add debug statements to the rubrik_GetFromAPI action, I can see it looping infinitely while getting 404 responses.

*Eventually* after waiting a long time (like until the next day) I can call this same workflow and watch it succeed against the same VM where it failed during the provisioning stage.

**Failure Information (for bugs)**

Please help provide information about the failure if this is a bug.
The workflow doesn't error out, and by default it does not echo any debug statements on the API calls themselves. If I modify the rubrik_GetFromAPI action to add a debug statement, I can see that the http response is a 404. It does this until the workflow is canceled.

Sometimes I can log into the Rubrik appliance and see that the SLA domain WAS set even though the API call is continuing to fail - this doesn't happen all of the time but it does happen on occasion.

* Use verbose outputs to capture any debug information.
Note that the "Rubrik status code" lines are my debug output. We see this issue even in environments with an unmodified action.

I truncated the log after a few 404s occurred but that same log line continues for about 15 minutes before I have an automated workflow kill it and email our backup team to assign an SLA Domain to the server.
```
[2019-06-21 12:08:51.271] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:51.283] [I] Rubrik-NON-PROD-DATACENTER:vmware/vm?primary_cluster_id=local&name=my_server_name:GET
[2019-06-21 12:08:52.048] [D] Rubrik status code: 200
[2019-06-21 12:08:52.049] [I] Refreshing all vCenters
[2019-06-21 12:08:52.050] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:52.060] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter:GET
[2019-06-21 12:08:52.609] [D] Rubrik status code: 200
[2019-06-21 12:08:52.611] [I] Refreshing vCenters : vCenter:::3f520a4e-7f36-4714-be1a-865ceb547fcd,vCenter:::93a43e49-e663-4496-8659-e4194d72c524,vCenter:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7
[2019-06-21 12:08:52.614] [I] Refreshing vCenter ID : vCenter:::3f520a4e-7f36-4714-be1a-865ceb547fcd
[2019-06-21 12:08:52.617] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:52.629] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/vCenter:::3f520a4e-7f36-4714-be1a-865ceb547fcd/refresh:POST
[2019-06-21 12:08:54.889] [D] Rubrik status code: 202
[2019-06-21 12:08:54.890] [I] Refreshing vCenter ID : vCenter:::93a43e49-e663-4496-8659-e4194d72c524
[2019-06-21 12:08:54.891] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:54.901] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/vCenter:::93a43e49-e663-4496-8659-e4194d72c524/refresh:POST
[2019-06-21 12:08:55.187] [D] Rubrik status code: 202
[2019-06-21 12:08:55.188] [I] Refreshing vCenter ID : vCenter:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7
[2019-06-21 12:08:55.189] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:55.199] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/vCenter:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7/refresh:POST
[2019-06-21 12:08:55.505] [D] Rubrik status code: 202
[2019-06-21 12:08:55.523] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:08:55.537] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:08:55.691] [D] Rubrik status code: 200
[2019-06-21 12:09:00.745] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:00.755] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:00.885] [D] Rubrik status code: 200
[2019-06-21 12:09:05.905] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:05.917] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:07.289] [D] Rubrik status code: 200
[2019-06-21 12:09:12.311] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:12.326] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:13.537] [D] Rubrik status code: 200
[2019-06-21 12:09:18.561] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:18.575] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:19.953] [D] Rubrik status code: 200
[2019-06-21 12:09:24.974] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:24.984] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:26.205] [D] Rubrik status code: 200
[2019-06-21 12:09:31.229] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:31.239] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:09:32.577] [D] Rubrik status code: 200
[2019-06-21 12:09:37.624] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:09:37.642] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:10:07.794] [D] Rubrik status code: 200
[2019-06-21 12:10:12.828] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:10:12.841] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:10:42.955] [D] Rubrik status code: 200
[2019-06-21 12:10:47.974] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:10:47.993] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:10:49.606] [D] Rubrik status code: 200
[2019-06-21 12:10:54.627] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:10:54.647] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:10:56.246] [D] Rubrik status code: 200
[2019-06-21 12:11:01.268] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:01.280] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:02.618] [D] Rubrik status code: 200
[2019-06-21 12:11:07.637] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:07.648] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:08.298] [D] Rubrik status code: 200
[2019-06-21 12:11:13.326] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:13.336] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:17.198] [D] Rubrik status code: 200
[2019-06-21 12:11:22.228] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:22.242] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:23.011] [D] Rubrik status code: 200
[2019-06-21 12:11:28.034] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:28.045] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:28.696] [D] Rubrik status code: 200
[2019-06-21 12:11:33.729] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:33.742] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:34.394] [D] Rubrik status code: 200
[2019-06-21 12:11:39.423] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:39.434] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:41.086] [D] Rubrik status code: 200
[2019-06-21 12:11:46.109] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:46.124] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:47.391] [D] Rubrik status code: 200
[2019-06-21 12:11:52.410] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:52.421] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:11:53.695] [D] Rubrik status code: 200
[2019-06-21 12:11:58.722] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:11:58.732] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:00.084] [D] Rubrik status code: 200
[2019-06-21 12:12:05.111] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:05.123] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:05.251] [D] Rubrik status code: 200
[2019-06-21 12:12:10.279] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:10.293] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:10.448] [D] Rubrik status code: 200
[2019-06-21 12:12:15.478] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:15.492] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:15.690] [D] Rubrik status code: 200
[2019-06-21 12:12:20.726] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:20.739] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:20.893] [D] Rubrik status code: 200
[2019-06-21 12:12:25.912] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:25.945] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:26.106] [D] Rubrik status code: 200
[2019-06-21 12:12:31.171] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:31.182] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:31.320] [D] Rubrik status code: 200
[2019-06-21 12:12:36.344] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:36.355] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:37.547] [D] Rubrik status code: 200
[2019-06-21 12:12:42.576] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:42.591] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:44.184] [D] Rubrik status code: 200
[2019-06-21 12:12:49.220] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:49.230] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:50.634] [D] Rubrik status code: 200
[2019-06-21 12:12:55.669] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:12:55.680] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:12:56.986] [D] Rubrik status code: 200
[2019-06-21 12:13:02.022] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:13:02.037] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:13:03.202] [D] Rubrik status code: 200
[2019-06-21 12:13:08.251] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:13:08.265] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:13:38.390] [D] Rubrik status code: 200
[2019-06-21 12:13:43.417] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:13:43.429] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:13.592] [D] Rubrik status code: 200
[2019-06-21 12:14:18.611] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:18.624] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:18.990] [D] Rubrik status code: 200
[2019-06-21 12:14:24.010] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:24.020] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:24.373] [D] Rubrik status code: 200
[2019-06-21 12:14:29.396] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:29.407] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:29.753] [D] Rubrik status code: 200
[2019-06-21 12:14:34.772] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:34.784] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:35.140] [D] Rubrik status code: 200
[2019-06-21 12:14:40.198] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:40.209] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:41.593] [D] Rubrik status code: 200
[2019-06-21 12:14:46.652] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:46.667] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:48.039] [D] Rubrik status code: 200
[2019-06-21 12:14:53.065] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:53.077] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:54.478] [D] Rubrik status code: 200
[2019-06-21 12:14:54.517] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:54.528] [I] Rubrik-NON-PROD-DATACENTER:vmware/vcenter/request/REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0:GET
[2019-06-21 12:14:55.868] [D] Rubrik status code: 200
[2019-06-21 12:14:55.869] [I] Request REFRESH_METADATA_cf9d09a1-1035-47d5-b55a-b7a75ff09de7_d2c87ff6-949b-4d95-92ff-0b3051466de5:::0 status is SUCCEEDED
[2019-06-21 12:14:55.908] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:55.951] [I] Rubrik-NON-PROD-DATACENTER:sla_domain?limit=9999&primary_cluster_id=local:GET
[2019-06-21 12:14:58.128] [D] Rubrik status code: 200
[2019-06-21 12:14:58.149] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:14:58.161] [I] Rubrik-NON-PROD-DATACENTER:vmware/vm?primary_cluster_id=local&limit=9999&name=my_server_name:GET
[2019-06-21 12:15:00.141] [D] Rubrik status code: 200
[2019-06-21 12:15:00.143] [I] VMID Found : VirtualMachine:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7-vm-4053
[2019-06-21 12:15:00.156] [I] Returning VirtualMachine:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7-vm-4053
[2019-06-21 12:15:00.170] [I] Looking up ReST Authorization for Rubrik-NON-PROD-DATACENTER
[2019-06-21 12:15:00.181] [I] Rubrik-NON-PROD-DATACENTER:vmware/vm/VirtualMachine:::cf9d09a1-1035-47d5-b55a-b7a75ff09de7-vm-4053:PATCH
[2019-06-21 12:15:01.783] [D] Rubrik status code: 404
[2019-06-21 12:15:03.457] [D] Rubrik status code: 404
[2019-06-21 12:15:05.135] [D] Rubrik status code: 404
[2019-06-21 12:15:06.780] [D] Rubrik status code: 404
[2019-06-21 12:15:08.378] [D] Rubrik status code: 404
[2019-06-21 12:15:09.933] [D] Rubrik status code: 404
```

**Steps to Reproduce**

Run the "Rubrik - Assign SLA Domain" workflow against a newly created vCenter VM.

* Use numbered list.

**Context**

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

vRO Version: 7.4
Rubrik Plugin version: I don't see a version, but I downloaded it from this project on 4/8/2019
Rubrik appliance version: 5.0
vCenter version: 6.5

Our Rubrik appliances are behind an F5 load balancer, though even going directly to the appliance yields the same result.

**Comment by pmilano1** [comment]
Greetings - can you send the JSON payload attached to the call that 404's, as well as the URI that is called? I've identified a bug related to the persistence that we will address.

**Comment by bender-the-greatest** [comment]
Here is the payload JSON: `{"configuredSlaDomainId":"undefined"}`

Obviously, "undefined" is the likely issue here. In this test case I am attempting to assign "DO NOT PROTECT" to the VM, which all documentation I've read so far indicates that we should be passing in "UNPROTECTED" to the API as the SLA domain in this case. However, the `rubrik_GetSlaId` action seems to be returning nothing to the `sla_id` attribute. Screenshot below of my most recent workflow run:

![image](https://user-images.githubusercontent.com/1764885/60125479-7f930f00-9751-11e9-9c24-7ce2887b4445.png)

I did a bit of digging in the [Rubrik Powershell module](https://github.com/rubrikinc/rubrik-sdk-for-powershell/blob/master/Rubrik/Private/Test-RubrikSLA.ps1#L19) and see that UNPROTECTED is used as the SLA ID, but NOT the SLA name. The ID is hardcoded in this case and triggered with a cmdlet switch so I don't see anySLA name reference in `Test-RubrikSLA`. Is there a different name I need to use with "Rubrik - Assign SLA Domain" for the explicit "Do Not Protect" policy?

For now I want to focus on getting the UNPROTECTED case working, as I learned yesterday that some of the SLA domains we've been attempting to assign actually don't exist yet in the environment where I've been testing, and this behavior can be explained if the SLA by name doesn't exist. I'm thinking I will probably need to duplicate and modify this workflow to bypass SLA ID resolution if there isn't an SLA name I can pass in to resolve, but I would like to avoid this approach if possible.

Additionally, "Rubrik - Assign SLA Domain" should probably fail the workflow if it can't resolve the SLA ID by name, because this otherwise results in an infinite loop (the longest I've let this run is ~1.5 hours).

**Comment by bender-the-greatest** [comment]
I did more testing and digging and cannot find a friendly name to reference "Do Not Protect" with when trying to resolve the "UNPROTECTED" SLA ID. I modified the `Library/Rubrik-DevOps/vRA-ResourceActions/Rubrik - Assign SLA Domain` workflow with two changes, and submitted a pull request:

1. Make sure the SLA Domain Name passed in exists, if not, throw an error
2. If "UNPROTECTED" is passed in as the SLA Domain name, bypass the SLA ID resolution and set the SLA ID to "UNPROTECTED".
